test/cuda/cufinufft_math_test.cu

```diff
@@ -0,0 +1,130 @@
+#include <complex>
+#include <iostream>
+#include <limits>
+#include <random>
+
+// Include the custom operators for cuComplex
+#include <cufinufft/contrib/helper_math.h>
+#include <cufinufft/types.h>
+
+// Helper function to create cuComplex
+template<typename T> cuda_complex<T> make_cuda_complex(T real, T imag) {
+  return cuda_complex<T>{real, imag};
+}
+
+// Helper function to compare cuComplex with std::complex<T> using 1 - ratio as error
+template<typename T>
+bool compareComplex(const cuda_complex<T> &a, const std::complex<T> &b,
```

Comment:
As we discussed today, this is too harsh a test, since when a complex number "lands" near the real axis, its imag part (by itself) may have high relative error, and that's ok. Instead of separately testing Re and Im rel errors, use cabs(a-b)/cabs(a)  < epsilon.    Ask if still doesn't make sense.